public/app/plugins/panel/state-timeline/timeline.ts

```diff
@@ -515,5 +514,14 @@ export function getConfig(opts: TimelineCoreOptions) {
 
 function getFillColor(fieldConfig: TimelineFieldConfig, color: string) {
   const opacityPercent = (fieldConfig.fillOpacity ?? 100) / 100;
-  return tinycolor(color).setAlpha(opacityPercent).toString();
+
+  if (color.length === 7 && color[0] === '#') {
+    color = color + Math.round(opacityPercent * 255).toString(16);
+  } else if (color.startsWith('rgb(')) {
+    // rgb() does not require the "a" suffix to accept alpha values in modern browsers:
+    // https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/rgb()#accepts_alpha_value
+    color = color.replace(')', `, ${opacityPercent})`);
+  }
+
+  return color;
```

Comment:
Can you try the performance of https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/themes/colorManipulator.ts#L242  ?  

If that is also bad, at we need a backup here to use tiny color or the above function for non hash colors

Reply:
not looking great

![image](https://user-images.githubusercontent.com/43234/120084554-d5d3ff80-c096-11eb-8226-d3ff5609dc54.png)

Reply:
ok, then maybe you can update that function? So we can share this logic / performance improvement for other use cases where we set alpha

Reply:
this part is problematic:

```
@returns A CSS color string. Hex input values are returned as rgb<br class="Apple-interchange-newline">
```

even if we just wanted to add a fast path here for the two cases i'm handling (rgb, hex), this return expectation means we _must_ do hex -> rgb parsing and conversion rather than simply appending alpha to the provided strings in both cases.

if we can make the return value match whatever is fed to it as the input format (either by default or via flag), then we can integrate the optimization, otherwise you can only optimize `rgb()` cases, while everything else must take the slow path.

Reply:
you can break that expectation, and return hex or rgb depending on what was passed